coverage/PENDING_SUBMIT/amdvlk/0595-COVERAGE-instruction-simplify-1406/cov-bitwise-shift-right-full-bits-no-effect-clamp.amber

```diff
@@ -0,0 +1,148 @@
+#!amber
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# A test for a coverage-gap found by the GraphicsFuzz project.
+
+# Short description: A fragment shader that covers specific LLVM code paths
+
+# The test passes because the shader always writes red.
+
+SHADER vertex variant_vertex_shader PASSTHROUGH
+
+# variant_fragment_shader is derived from the following GLSL:
+# #version 320 es
+#
+# precision highp float;
+# precision highp int;
+#
+# layout(set = 0, binding = 0) uniform buf0
+# {
+#     int one;
+# };
+#
+# layout(location = 0) out vec4 _GLF_color;
+#
+# void main()
+# {
+#     // Arithmetic shift replicates the sign bit.
+#     int a = 0xffffffff >> one;
+#     // Clamps to 2.0.
+#     float f = clamp(5.0, 1.0, a == 0xffffffff ? 2.0 : 3.0);
```

Comment:
I'd add more parentheses to make sure the a == 0xff ? 2 : 3 means (a == 0xff) ? 2 : 3 instead of a == (0xff ? 2 : 3).